docs/spacefinder.md

```diff
@@ -0,0 +1,85 @@
+# Spacefinder
+
+Spacefinder identifies slots in articles where ads can be inserted dynamically. Every paragraph is assessed against a set of rules, with 'winning' candidates having ads placed above them.
+
+- [How Spacefinder works](#how-spacefinder-works)
+- [Spacefinder rules](#spacefinder-rules)
+- [Style and layout of dynamic ads](#style-and-layout-of-dynamic-ads)
+
+Ideally Spacefinder hits the sweet spot between high ad ratio for us and a peerless reading experience for _Guardian_ website visitors.
+
+---
+
+## How Spacefinder works
+
+Some ad slots on the website are fixed. Come rain or shine, they're there. These include:
+
+- The banner slot above the navbar on desktop
+- A space to the top right of articles on desktop
+
+By contrast, ad slots identified by Spacefinder vary from page to page. A short article may only have space for one, while a long read ought to fit many more.
+
+Once a page has loaded, Spacefinder cycles through the paragraphs inside the `article-body-commercial-selector` container, highlighted below. (The Adobe / Billie Eilish ad is an example of an ad inserted by Spacefinder.)
+
+![Article body section of a Guardian article](assets/spacefinder-article-body-example.png)
+
+Within this space, ‘candidate’ paragraphs are assessed in relation to the last ‘winner’. If the candidate paragraph is far enough away from the last winning paragraph - and from any other specified element types - an ad can be added in front of it. If not, Spacefinder moves on to the next paragraph.
+
+![Visual explainer of how Spacefinder tests for ad slots](assets/spacefinder-process-visualised.png)
+
+The space between paragraphs is calculated using the CSS ‘top’ property. There are too many moving parts (e.g. filtering for specific elements) for there to be a catch-all equation that sums up what Spacefinder does, but the gist of it is:
+
+***If the distance between the top of the last winning paragraph and the top of the candidate paragraph is greater than or equal to the maximum potential ad height plus padding, the candidate paragraph qualifies for an ad.***
+
+Or to simplify it further:
+
+![Pseudo equation explaining Spacefinder logic in a nutshell](assets/spacefinder-equation.png)
+
+I have made up the value names above to try to boil the essentials down - they don’t match what’s going on in the code itself. The ad heights mentioned are pulled in from `ad-sizes.js`.
+
+---
+
+[ [Back to top &#8593;](#spacefinder) ]
+
+---
+
+## Spacefinder rules
+
+Rules are currently set in [`article-body-adverts.js`](https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/commercial/modules/article-body-adverts.js) in the frontend repository. The desktop configuration sits inside `addDesktopInlineAds`, which has two sets of rules. The mobile configuration is inside `addMobileInlineAds`.
+
```

Comment:
If you use links, it’s sometimes worth extracting them like this.

Also, you probably want to pin to a permalink, which would be https://github.com/guardian/frontend/blob/cd3dc40766710a8931357c1fa50a5a5c7c6de961/static/src/javascripts/projects/commercial/modules/article-body-adverts.js, in case the final file ever moves.

```suggestion
Rules are currently set in [`article-body-adverts.js`][] in the frontend repository. The desktop configuration sits inside `addDesktopInlineAds`, which has two sets of rules. The mobile configuration is inside `addMobileInlineAds`.

[`article-body-adverts.js`]: https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/commercial/modules/article-body-adverts.js

```